semgrep-core/src/engine/Match_tainting_rules.ml

```diff
@@ -173,17 +175,20 @@ let lazy_force x = Lazy.force x [@@profiling]
 (*****************************************************************************)
 
 let check_bis ~match_hook (default_config, equivs)
-    (taint_rules : (Rule.rule * Rule.taint_spec) list) file lang ast =
+    (taint_rule : Rule.rule * Rule.taint_spec) file lang ast =
+  (* @Iago I went and modified this to work one rule at a time *)
```

Comment:
@IagoAbal made some changes here so this runs one rule at a time, possibly there are smarter things to do but I just made the dumbest possible change